### ISSUE TITLE
Blender jupyter notebook-kernel error

### ISSUE THREAD

**Issue by bosse89** [Hjorthmedh/Snudda#175]
When attempting to follow the example in:
https://github.com/Hjorthmedh/Snudda/blob/master/examples/notebooks/blender_example.ipynb
I get an error when I load the jupyter notebook. This error is due to some error in another repository: https://github.com/cheng-chi/blender_notebook
However, the Snudda blender examples should ideally refer to examples that work. I might of course have made some misstake. However, I did get the same error when attempting in a ubuntu virtual machine in virtualbox.
The following is the error when running jupyter notebook (in Windows 10) and then starting the blender-notebook kernel:

################################################################################
...
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\Users\Bosse\AppData\Local\Temp\tmpvqy59jpl\kernel.py", line 30, in <module>
    from ipykernel.kernelapp import IPKernelApp
  File "C:\Users\Bosse\AppData\Roaming\Python\Python37\site-packages\ipykernel\__init__.py", line 2, in <module>
    from .connect import *
  File "C:\Users\Bosse\AppData\Roaming\Python\Python37\site-packages\ipykernel\connect.py", line 18, in <module>
    import jupyter_client
  File "C:\Users\Bosse\AppData\Roaming\Python\Python37\site-packages\jupyter_client\__init__.py", line 4, in <module>
    from .connect import *
  File "C:\Users\Bosse\AppData\Roaming\Python\Python37\site-packages\jupyter_client\connect.py", line 21, in <module>
    import zmq
  File "d:\programs\anaconda3\lib\site-packages\zmq\__init__.py", line 125, in <module>
    from zmq import backend
  File "d:\programs\anaconda3\lib\site-packages\zmq\backend\__init__.py", line 32, in <module>
    raise original_error from None
  File "d:\programs\anaconda3\lib\site-packages\zmq\backend\__init__.py", line 27, in <module>
    _ns = select_backend(first)
  File "d:\programs\anaconda3\lib\site-packages\zmq\backend\select.py", line 32, in select_backend
    mod = import_module(name)
  File "D:\programs\Blender Foundation\Blender 2.93\2.93\python\lib\importlib\__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "d:\programs\anaconda3\lib\site-packages\zmq\backend\cython\__init__.py", line 6, in <module>
    from . import (
ImportError: cannot import name 'constants' from partially initialized module 'zmq.backend.cython' (most likely due to a circular import) (d:\programs\anaconda3\lib\site-packages\zmq\backend\cython\__init__.py)
[W 13:00:19.728 NotebookApp] Timeout waiting for kernel_info reply from 4725f5dd-8cde-4e7c-9229-d791cab0629f
################################################################################

**Temporary solution:**
Not sure, but you might get it to work if you've installed an older Blender version such as 2.79 or lower.

**Solution:**
No solution yet. However here are some notes:
Even if you would install Blender manually as a Python module (https://wiki.blender.org/wiki/Building_Blender/Other/BlenderAsPyModule), there is currently a bug (I think) in Snudda such that when installed using 'pip3 install snudda', the Blender folder (Snudda/snudda/plotting/Blender) containing all the visualization code used in the Snudda Blender example (blender_example.ipynb) is missing. If you add it manually, it will not work:
#################################################################
Traceback (most recent call last):
  File "D:\code\python\pycharmProjects\testBlender2\test.py", line 11, in <module>
    vn.visualise(neuron_id=[0])
  File "D:\Python\Python39\lib\site-packages\snudda\plotting\Blender\visualisation\visualise_network.py", line 80, in visualise
    mat_msd1.diffuse_color = (77. / 255, 151. / 255, 1.0)
ValueError: bpy_struct: item.attr = val: sequences of dimension 0 should contain 4 items, not 3
#################################################################

This is most likely because of the new Blender version (2.8x+) in which there was a lot of changes.

Furthermore, the SWC import addon (https://github.com/MartinPyka/SWC2Blender) does not work due to the new Blender version. I've made a new addon which I'm currently testing.

**Comment by Hjorthmedh** [comment]
Great, thanks for looking into this. My Debian linux is still running the old version of Blender last I checked. Let me know how it goes, and once Debian updates to the new version I can check also.

**Comment by bosse89** [comment]
I should say that I got the Blender 2.93 working last week or so by installing Blender as a python module and converting the addon to be part of the visualize_network code. Files and example with instructions are on the way.

**Comment by Hjorthmedh** [comment]
Hej Bo, any update on the Blender fix for 2.93?

**Comment by bosse89** [comment]
Hej I missed this comment. Files were added in the latest pull request we did together
Example code for how to use the VisualizeNetwork2 class.
https://github.com/Hjorthmedh/Snudda/tree/master/examples/python-scripts/blender-example
The VisualizeNetwork2 class is with the previous one:
https://github.com/Hjorthmedh/Snudda/blob/master/snudda/plotting/Blender/visualisation/visualise_network2.py
Planning to add some instructions on the example page and perhaps a link in the documentation. Will check with you first before pushing.
